src/_cffi_src/utils.py

```diff
@@ -80,9 +82,18 @@ def build_ffi(module_name, cdef_source, verify_source, libraries=[],
 
 
 def extra_link_args(platform):
-    if platform != "win32":
+    if platform != "win32" or not _compiler_is_msvc():
```

Comment:
Good point. I think I can refactor this significantly so `extra_link_args` takes the compiler_type rather than platform since that's what we actually care about here.